pyk/src/tests/unit/test_kcfg.py

```diff
@@ -849,3 +850,69 @@ def test_no_cell_rewrite_to_dots() -> None:
 
     result = no_cell_rewrite_to_dots(term)
     assert result == term
+
+
+CREATE_SPLIT_BY_NODES_TEST_DATA: Final = (
+    (CTerm.bottom(), [CTerm.top(), CTerm.bottom()], None),
+    (CTerm.top(), [CTerm.top(), CTerm.bottom()], None),
+    (
+        CTerm.top(),
+        [CTerm(config('X')), CTerm(config('Y'))],
+        None,  # todo: support split from top, because top means anything can be matched
+    ),
+    (
+        CTerm.top(),
+        [CTerm.top(), CTerm.top()],
+        KCFG.Split(
+            KCFG.Node(1, CTerm.top()), [(KCFG.Node(2, CTerm.top()), CSubst()), (KCFG.Node(3, CTerm.top()), CSubst())]
+        ),
+    ),
```

Comment:
This one violates that the target of the splits are mutually exclusive, since nodes 2 and 3 overlap.

Reply:
I think here, we should just put a comment on the ones that violate this condition, and otherwise move forward. Eventually we should enforce that split nodes are actually semantically splits, and then the comments you put here will be useful to know that these tests can be deleted/ignored.

Reply:
I have left some comments for these test case.